### ISSUE TITLE
Поиск в массиве, элементы которого объекты(json) 

### ISSUE THREAD

**Issue by vitalikost** [redexp/geekhub-2016-js#11]
Добрый день! Нужна идея,или направить в нужное русло. Есть массив полученный запросом из сервера, элементы массива  json объекты. Пример кода:
``` javascript
var tabl = [];
for(i=0;i<10;i++){ 
var a={}; 
a.kod = 100+i; 
a.name = "Element "+(100+i);
 tabl.push(a);}
//Результат:
//[Object, Object, Object, Object, Object, Object, Object, Object, Object, Object]
//tabl[0]
//Object {kod: 100, name: "Element 100"}
//tabl[5]
//Object {kod: 105, name: "Element 105"}
```

Как правильно организовать поиск элемента, допустим с значением kod=105;
Прямой перебор каждый раз не годиться. Записей порядка 17000. 
Спасибо!

**Comment by rocko-n** [comment]
Можно вместо массива использовать объект:
```JS
var tabl = {};
for(var i=0;i<10;i++){ 
var a={}; 
a.kod = 100+i; 
a.name = "Element "+(100+i);
 tabl[a.kod] = a;
}
console.log(tabl.hasOwnProperty(105));
```

**Comment by vitalikost** [comment]
Нашел то искал:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find
``` javascript
function find_elem(elem) { 
    return elem.kod === 105;
}
tabl.find(find_elem);
//Object {kod: 105, name: "Element 105"}
```

**Comment by redexp** [comment]
правильный ответ - искать должен сервер и возвращать нужное значение. Ты шо с ума сошёл 17000 объектов в браузер выплёвывать

**Comment by vitalikost** [comment]
Эм, это уже отфильтрованный запрос, первоначально было больше 1 млн.

**Comment by redexp** [comment]
ищи решение

**Comment by vitalikost** [comment]
Ну тут задача такая, если кратко: Есть ПО(Рабочее место кассира в маркете) работает БД MS SQL. К базе с я с помощью ноды уже подключился. Все движение и нажатие ПО фиксирует в БД. 
Задача: Организовать возможность просмотра этих действий охране:

systemid	EVENTCODE	EVENTTIME	EVENTCONTENT	EVENTDATA

2	10013	20161231080640	Попытка открытия чека.	0:1:
2	10010	20161231080641	Попытка добавления товара в чек.	1:1:2:0207296000428
2	10014	20161231080641	Открыт чек.	1:1:
2	10318	20161231080700	Был рассчитаны скидки.	1:1:
2	10011	20161231080700	Добавлен товар в чек.	1:1:2:7296:1:7296:44212:42:0
2	10011	20161231080703	Добавлен товар в чек.	1:1:2:6996:1:6996:325:446:0
2	10010	20161231080703	Попытка добавления товара в чек.	1:1:2:0206996004460
2	10010	20161231080703	Попытка добавления товара в чек.	1:1:2:4015400437932
2	10318	20161231080703	Был рассчитаны скидки.	1:1:
2	10318	20161231080704	Был рассчитаны скидки.	1:1:
2	10011	20161231080704	Добавлен товар в чек.	1:1:2:134811:1:134811:2695:1:1
2	10025	20161231080709	Попытка входа в режим оплаты чека.	1:1:
2	10026	20161231080709	Вход в режим оплаты чека.	1:1:
2	10318	20161231080709	Был рассчитаны скидки.	1:1:
2	10031	20161231080713	Попытка оплаты чека.	1:1:0:5000:0
2	10072	20161231080714	Открытие денежного ящика.	
2	10071	20161231080714	Попытка открытия денежного ящика.	
2	10317	20161231080715	Начало печати чека.	1:1:
2	10032	20161231080720	Оплата чека.	1:1:0:5000:0
2	10071	20161231080721	Попытка открытия денежного ящика.	
2	10031	20161231080721	Попытка оплаты чека.	1:1:0:0:0
2	10072	20161231080721	Открытие денежного ящика.	
2	10317	20161231080722	Начало печати чека.	1:1:
2	10032	20161231080722	Оплата чека.	1:1:0:0:0
2	10194	20161231080725	Попытка закрытия чека.	1:1:2
2	10195	20161231080729	Чек закрыт.	1:1:2

Во такой бред я им могу зараз вывести, их это не устраивает.
Вся важная инфа в последнем поле, в базе это 1 большое текстовое поле.
Вот его я и хочу уже парсить.

**Comment by redexp** [comment]
Если им надо вывести много данных, то выводи их пачками, по 10 например, сделай [пагинатор](http://getbootstrap.com/components/#pagination) и хай они смотрят что захотят. Добавь форму поиска по словам, по дате, и хай ищут что им нужно

**Comment by vitalikost** [comment]
Так, это уже идея. В MS SQL чтобы ограничить количество записей есть Оператор  TOP.
Вот как брать их пачками, это зараз буду искать. Спасибо за идею.

**Comment by vitalikost** [comment]
Может кому надо будет. В MS SQL нет команды LIMIT, которой можно вывести диапазон выборки.
В просторах нашел аналог:

```
  WITH EVENTS_ AS  
(  
    SELECT *,  
    ROW_NUMBER() OVER (ORDER BY [EVENTCODE]) AS RowNumber  
    FROM [test].[dbo].[EVENTS]
)   
SELECT *    
FROM EVENTS_   
WHERE RowNumber BETWEEN 50 AND 60;  

```

Принцип роботы прост. Добавляем поле RowNumber к выборке по таблице EVENTS, а потом по нем фильтруем.
